File: 백준/20444/solution.py
import sys
input = sys.stdin.readline
N, K = map(int, input().split())

left, right = 0, N // 2 + 1
while left < right:
    mid = left + (right - left) // 2
    cuts = (mid+1)*(N-mid+1)
    if cuts == K:
        print('YES')
        exit(0)
    elif cuts < K:
        left = mid + 1
    else:
        right = mid
print('NO')

File: 백준/20444/solution.swift
import Foundation
let nk = readLine()!.split(separator: " ").compactMap{Int($0)}
let N = nk[0]
let K = nk[1]

func binarySearch(_left: Int, _right: Int) -> Bool {
    var left = _left
    var right = _right

    while left < right {
        let mid = left + (right - left) / 2
        let cuts = (mid+1) * (N-mid+1)

        if (cuts == K) {
            return true 
        } else if(cuts < K) {
            left = mid + 1
        } else {
            right = mid
        }
    }
    return false
}

let answer = binarySearch(_left: 0, _right: N/2 + 1)
print(answer == true ? "YES" : "NO")